compiler/noirc_evaluator/src/acir/tests/instructions.rs

```diff
@@ -1 +1,302 @@
+    public parameters: []
+    return values: []
+    BRILLIG CALL func: 0, inputs: [w0 - w1], outputs: [w2]
+    ASSERT 0 = w0*w2 - w1*w2 - 1
+
+    unconstrained func 0: directive_invert
+    0: @21 = const u32 1
+    1: @20 = const u32 0
+    2: @0 = calldata copy [@20; @21]
+    3: @2 = const field 0
+    4: @3 = field eq @0, @2
+    5: jump if @3 to 8
+    6: @1 = const field 1
+    7: @0 = field field_div @1, @0
+    8: stop &[@20; @21]
+    ");
+}
+
+#[test]
+fn cast() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: u8):
+        v1 = cast v0 as Field
+        return v1
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    // cast is a no-op
+    // (casting in Noir might also involve a truncate operation if it's casting from
+    // a larger integer type to a smaller one, but casting in SSA is just changing types)
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0]
+    public parameters: []
+    return values: [w1]
+    BLACKBOX::RANGE input: w0, bits: 8
+    ASSERT w1 = w0
+    ");
+}
+
+#[test]
+fn call_without_predicate() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: Field):
+        v1 = call f1(v0) -> Field
+        return v1
+    }
+
+    acir(fold) fn one f1 {
+      b0(v0: Field):
+        v1 = add v0, Field 1
+        return v1
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0]
+    public parameters: []
+    return values: [w1]
+    CALL func: 1, predicate: 1, inputs: [w0], outputs: [w2]
+    ASSERT w2 = w1
+
+    func 1
+    private parameters: [w0]
+    public parameters: []
+    return values: [w1]
+    ASSERT w1 = w0 + 1
+    ");
+}
+
+#[test]
+fn call_with_predicate() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: Field, v1: u1):
+        enable_side_effects v1
+        v2 = call f1(v0) -> Field
+        return v2
+    }
+
+    acir(fold) fn one f1 {
+      b0(v0: Field):
+        v1 = add v0, Field 1
+        return v1
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0, w1]
+    public parameters: []
+    return values: [w2]
+    BLACKBOX::RANGE input: w1, bits: 1
+    CALL func: 1, predicate: w1, inputs: [w0], outputs: [w3]
+    ASSERT w3 = w2
+
+    func 1
+    private parameters: [w0]
+    public parameters: []
+    return values: [w1]
+    ASSERT w1 = w0 + 1
+    ");
+}
+
+#[test]
+fn not() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: u8):
+        v1 = not v0
+        return v1
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0]
+    public parameters: []
+    return values: [w1]
+    BLACKBOX::RANGE input: w0, bits: 8
+    ASSERT w1 = -w0 + 255
+    ");
+}
+
+#[test]
+fn truncate_u16_to_6_bits() {
```

Comment:
What do you think about a test that compares the output of a truncate and a div. e.g., `truncate v0 to 6 bits, max_bit_size: 8` vs. `div v0, u16 64`. There would start to be some differences when it comes to different predicates (we can just use predicates of one to start) and bit sizes when truncating fields. This may be better for a follow-up, but just taking note.